File: frontend_better_days/lib/models/journal.dart
import 'package:intl/intl.dart';

class Journal {
  final String id;
  final String title;
  final String content;
  final List<String> tags;
  final DateTime? createdAt;
  final Map<String,dynamic>? createdBy;

  Journal({
    required this.id,
    required this.title,
    required this.content,
    required this.tags,
    required this.createdAt,
    required this.createdBy,
  });

  factory Journal.fromJson(Map<String, dynamic> json) {
    return Journal(
      createdBy: json['created_by'],
      id: json['id'] ?? "",
      title: json['title'] ?? "",
      content: json['content'] ?? "",
      tags: List<String>.from(json['tags'] ?? []),
      createdAt: DateTime.tryParse(json['created_at'] ?? ""),
    );
  }

  String get formattedDate {
    if(createdAt==null) return "";
    return DateFormat("dd MMM yyyy hh:mm a").format(createdAt!);
  }
}

File: backend/app/models/journal.py
from sqlalchemy import Column, Integer, String, JSON, DateTime

from app.database.connection import Base


class Journal(Base):
    __tablename__ = "journals"

    id= Column(Integer, primary_key=True, index=True,nullable=False)
    content= Column(String,nullable=True)
    title= Column(String,nullable=True)
    created_by= Column(String,nullable=False)
    tags = Column(JSON,nullable=True)
    created_at= Column(DateTime,nullable=False)

    class Config:
        from_attributes = True